src/libcadet/model/parts/TwoDimensionalConvectionDispersionOperator.cpp

```diff
@@ -850,6 +850,8 @@ bool TwoDimensionalConvectionDispersionOperator::configure(UnitOpIdx unitOpIdx,
 	else
 		registerParam2DArray(parameters, _velocity, [=](bool multi, unsigned int sec, unsigned int compartment) { return makeParamId(hashString("VELOCITY"), unitOpIdx, CompIndep, compartment, BoundStateIndep, ReactionIndep, multi ? sec : SectionIndep); }, _nRad);
 
+	std::vector<int> _dir(_nRad, 1);
```

Comment:
This creates a local variable that shadows the member variable with the same name.
You probably wanted to do
```cpp
_dir = std::vector<int>(_nRad, 1);
```

You should also initialize the vector in the constructor with size `0` to prevent unnecessary memory allocation.

Reply:
ah yes, I was a bit confused there...

Reply:
> You should also initialize the vector in the constructor with size `0` to prevent unnecessary memory allocation.

Sorry, can you tell me where would that be?

Reply:
Around line 656 in `TwoDimensionalConvectionDispersionOperator::TwoDimensionalConvectionDispersionOperator()`.
There's the syntax
```cpp
SomeClass::SomeClass() : _var1(value), _var2(ctorArg1, ctorArg2) { /* ... /* }
```
The part between `:` and `{` is an *initializer list*. Here, we can initialize (or call the constructor) of all our member variables. But we can also leave out the ones we don't want to initialize here. This happens before the block `{ }` is executed. Note that the variables that do appear have to be in the order of declaration in the class.

My suggestion is to add `_dir(0)` to this list. This creates a vector of size `0`, which should prevent memory allocation.

Reply:
Thanks; How do you decide what to initialize? E.g. `_curVelocity` is not initialized there. 

Also, should I resize it like other radially dependent parameter vectors are?
https://github.com/modsim/CADET/blob/0ecc57660ae38712f0715a82bdb15ecb1b7ef587/src/libcadet/model/parts/TwoDimensionalConvectionDispersionOperator.cpp#L731

Reply:
You don't need to `resize()` since you replace the whole vector later on (when you assign a new vector like in `_dir = std::vector<int>(...)`.

It would be consistent to also initialize the other vectors with size `0`. I was a little lazy, obviously. It doesn't matter much, though. Memory allocation only happens at the beginning of a simulation and we're talking about small sizes. I just wanted to raise awareness for memory stuff. 😄